test/channelz_test.go

```diff
@@ -1196,6 +1200,53 @@ func TestCZServerSocketMetricsKeepAlive(t *testing.T) {
 	}
 }
 
+func TestCZSocketGetSecurityValueTLS(t *testing.T) {
+	defer leakcheck.Check(t)
+	channelz.NewChannelzStorage()
+	e := tcpTLSRREnv
+	te := newTest(t, e)
+	te.startServer(&testServer{security: e.security})
+	defer te.tearDown()
+	te.clientConn()
+	if err := verifyResultWithDelay(func() (bool, error) {
+		tchan, _ := channelz.GetTopChannels(0)
+		if len(tchan) != 1 {
+			return false, fmt.Errorf("there should only be one top channel, not %d", len(tchan))
+		}
+		if len(tchan[0].SubChans) != 1 {
+			return false, fmt.Errorf("there should only be one subchannel under top channel %d, not %d", tchan[0].ID, len(tchan[0].SubChans))
+		}
+		var id int64
+		for id = range tchan[0].SubChans {
+			break
+		}
+		sc := channelz.GetSubChannel(id)
+		if sc == nil {
+			return false, fmt.Errorf("there should only be one socket under subchannel %d, not 0", id)
+		}
+		if len(sc.Sockets) != 1 {
+			return false, fmt.Errorf("there should only be one socket under subchannel %d, not %d", sc.ID, len(sc.Sockets))
+		}
+		for id = range sc.Sockets {
+			break
+		}
+		skt := channelz.GetSocket(id)
+		cert, _ := tls.LoadX509KeyPair(testdata.Path("server1.pem"), testdata.Path("server1.key"))
+		securityVal := skt.SocketData.Security.(*credentials.TLSChannelzSecurityValue)
```

Comment:
Do an `ok` check to avoid panic is `Security` is not the expected type.

Reply:
done